datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -377,12 +378,15 @@ private synchronized boolean waitForEventThreadToJoin() {
   // Waits for a notification for specified duration from the event thread before acquiring the Coordinator object.
   private synchronized void waitForNotificationFromEventThread(Duration duration) {
     try {
-      // This intrinsic conditional variable helps to halt threads (zk callback threads, main server thread) before
+      // This intrinsic conditional variable(CV) helps to halt threads (zk callback threads, main server thread) before
       // attempting to acquire the Coordinator object. We never halt the event thread (coordinator thread)
       // explicitly via this CV.
       _log.info("Thread {} will wait for notification from the event thread for {} ms.",
           Thread.currentThread().getName(), duration.toMillis());
-      this.wait(duration.toMillis());
+      while (!_notified) {
```

Comment:
this is worry some if event thread does not get free for 60 mins, we will keep be waiting which will trigger timeout on the deployment framework.

Reply:
True but what alternate choice do we have ? Hopefully, this doesn’t happen often. Is the heaviest operation partition assignment ? If it is, then it should now take < 10 secs on an average ?

Reply:
in this case, I would just have one time wait, something like

if (!_notified) {
this.wait(duration.toMillis());
}

bcs, all we want is to let the evenThread enter into the handleEvent, we added all this logic for that purpose only, right?

Reply:
Are you suggest changing while to if condition ? While is there to handle other spurious signals and not via notify all